### ISSUE TITLE
[Feature]: Fail if `logprobs=True` and provider doesn't support them

### ISSUE THREAD

**Issue by jamesbraza** [BerriAI/litellm#4031]
### The Feature

Currently with `LiteLLM==1.40.3`, I invoke `litellm.acompletion(..., model="claude-3-opus-20240229", logprobs=True, top_logprobs=1, ...)`.

In the response, despite passing `logprob=True`, there are no logprobs present. This is because:
1. Anthropic doesn't yet support logprobs (I just learned this today myself)
2. LiteLLM will silently just not give `logprobs`, even though `logprobs=True` was passed

### Motivation, pitch

I think it's unexpected that when one passes `logprobs=True` that no `logprobs` can be given back. I would rather the code blow up with a `ValueError("API provider XYZ doesn't support logprobs.")`

In other words, if a provider doesn't use `logprobs`, and `logprobs=True` is passed to `litellm.acompletion`, the function show blow up with a `ValueError`

**Comment by ishaan-jaff** [comment]
We already do this @jamesbraza 

try this 

```python


def test_litellm_anthropic_with_logprobs_exception():
    try:
        import litellm
        litellm.set_verbose = True
        litellm.drop_params = False

        litellm.completion(
            model="claude-3-opus-20240229",
            messages=[{"role": "user", "content": "What is the meaning of life?"}],
            logprobs=True,
            top_logprobs=1,
        )
        pytest.fail("Request should have failed - bad api key")
    except Exception as e:
        print("exception: ", e)

```

<img width="1250" alt="Screenshot 2024-06-05 at 9 37 52 PM" src="https://github.com/BerriAI/litellm/assets/29436595/627941bc-8630-43b7-80ed-73a9df4e0e5e">

**Comment by ishaan-jaff** [comment]
feel free to re-open if I did not address this

**Comment by jamesbraza** [comment]
Oh thanks @ishaan-jaff nice and I appreciate the quick response! Sorry for the false positive request
